src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1131,12 +1133,25 @@ void OrbitMainWindow::closeEvent(QCloseEvent* event) {
 
 void OrbitMainWindow::SetTarget(const orbit_qt::StadiaTarget& target) {
   const orbit_qt::StadiaConnection* connection = target.GetConnection();
-  orbit_qt::ServiceDeployManager* service_deploy_manager = connection->GetServiceDeployManager();
+  ServiceDeployManager* service_deploy_manager = connection->GetServiceDeployManager();
   app_->SetSecureCopyCallback([service_deploy_manager](std::string_view source,
                                                        std::string_view destination) {
     CHECK(service_deploy_manager != nullptr);
     return service_deploy_manager->CopyFileToLocal(std::string{source}, std::string{destination});
   });
+
+  QObject::connect(service_deploy_manager, &ServiceDeployManager::socketErrorOccurred, this,
+                   [this, instance_name = target.GetConnection()->GetInstance().display_name](
+                       std::error_code error) {
+                     QMessageBox::critical(
+                         this, "Connection error",
+                         QString("The connection to instance \"%1\" failed with error message: %2")
+                             .arg(instance_name)
+                             .arg(QString::fromStdString(error.message())),
+                         QMessageBox::Ok);
+                     QApplication::exit(kEndSessionReturnCode);
+                   });
```

Comment:
As far as I can see this connection could technically be made twice between the same objects, when SetTarget is called twice. So I would add the Qt::UniqueConnection flag. https://doc.qt.io/qt-5/qt.html#ConnectionType-enum

Reply:
I did not know about those. Good idea, I added it.